src/compiler/binder.ts

```diff
@@ -1574,6 +1574,11 @@ namespace ts {
             function onExit(node: BinaryExpression, state: WorkArea) {
                 if (!state.skip) {
                     const operator = node.operatorToken.kind;
+
+                    if (operator === SyntaxKind.CommaToken) {

```

Comment:
Why would this be in `onExit` and not `onRight`?

Reply:
If you look up a bit, the `left` is processed in `onOperator`, presumably because there's some work that is completed after `onLeft` is completed. That's how it was added in #44487.

I tried putting it into `onRight` first and hit infinite recursion, then realized that the other call wasn't in `onLeft` and moved it down.

Reply:
Perhaps I can get away with moving these around, though, and try running them after `maybeBind` in `onLeft` and `onRight` instead. I'll try it. It'd sure look less weird.

Reply:
No, unfortunately this doesn't seem to work:

```ts
            function onLeft(left: Expression, state: WorkArea, _node: BinaryExpression) {
                if (!state.skip) {
                    const node = maybeBind(left);
                    if (left.kind === SyntaxKind.CommaToken) {
                        maybeBindExpressionFlowIfCall(left);
                    }
                    return node;
                }
            }
```

I don't quite understand the mechanics of the `BinaryExpressionStateMachine` thing, but it seems like `maybeBindExpressionFlowIfCall` needs to happen after `onLeft` and `onRight` have fully returned.

Reply:
Ignore this, I'm a fool and am checking the wrong node's kind.

Reply:
I've moved things around to be a little less surprising.